### PR TITLE
ProtoDC2 v2.1.2 Reader

### PR DESCRIPTION
**Pull request by dkorytov**
ProtoDC2 Version 2.1.2 Update

a) Galaxy arc second size has been fixed. 
b) The GCR 'size_disk_true'/'size_bulge_true' now point to arc second sizes (as indicated by the descqa schema) instead of mpc sizes. 
c) Ellipticities have been updated. I need to do a few more test/check with people before I would declare them to be fine.  
d) Added a new yaml file for v2.1.2 and updated the alias of protoDC2.yaml. 
f) Position angle extends to 180 degrees. Before was 90 deg.

### CODE REVIEW

File: GCRCatalogs/alphaq.py [hunk 1]
```diff
@@ -66,22 +66,26 @@ def _subclass_init(self, filename, **kwargs):
...
-            'disk_sersic_index':  'morphology/diskSersicIndex',
-            'bulge_sersic_index': 'morphology/spheroidSersicIndex',
-            'position_angle':     (lambda pos_angle: np.rad2deg(np.rad2deg(pos_angle)), 'morphology/positionAngle'), #I converted the units the wrong way, so a double conversion is required.
-            'ellipticity_1':      (lambda ellip2, pos_angle: ellip2/np.tan(2*np.rad2deg(pos_angle)), 'morphology/totalEllipticity2', 'morphology/positionAngle'), # Two points: pos_angle needs to be converted to radians from the over converted value. ellip_1 should equal |ellip|*cos(2*pos_angle) but was assigned |ellip|*sin(2*pos_angle)
-            'size_true':          (lambda size1, size2, lum1, lum2: ((size1*lum1)+(size2*lum2))/(lum1+lum2), 'morphology/diskHalfLightRadius', 'morphology/spheroidHalfLightRadius', 'LSST_filters/diskLuminositiesStellar:LSST_r:rest', 'LSST_filters/spheroidLuminositiesStellar:LSST_r:rest'),
+            'size_disk_true':     'morphology/diskHalfLightRadiusArcsec',
+            'size_bulge_true':    'morphology/spheroidHalfLightRadiusArcsec',
+            'disk_Sersic_index':  'morphology/diskSersicIndex',
```
Comment (yymao) [inline GCRCatalogs/alphaq.py]: change name to `sersic_disk` to follow the [new schema](https://github.com/LSSTDESC/gcr-catalogs/blob/add-schema/GCRCatalogs/SCHEMA.md)

File: GCRCatalogs/alphaq.py [hunk 2]
```diff
@@ -66,22 +66,26 @@ def _subclass_init(self, filename, **kwargs):
...
+            'bulge_Sersic_index': 'morphology/spheroidSersicIndex',
```
Comment (yymao) [inline GCRCatalogs/alphaq.py]: change name to `sersic_bulge` to follow the [new schema](https://github.com/LSSTDESC/gcr-catalogs/blob/add-schema/GCRCatalogs/SCHEMA.md)

File: GCRCatalogs/alphaq.py [hunk 3]
```diff
@@ -66,22 +66,26 @@ def _subclass_init(self, filename, **kwargs):
...
             'position_z':         'z',
             'velocity_x':         'vx',
             'velocity_y':         'vy',
             'velocity_z':         'vz',
         }
-
+        if catalog_version < StrictVersion('2.1.2'):
+            self._quantity_modifiers.update({
+                'position_angle':     (lambda pos_angle: np.rad2deg(np.rad2deg(pos_angle)), 'morphology/positionAngle'), #I converted the units the wrong way, so a double conversion is required.
+            })
```
Comment (yymao) [inline GCRCatalogs/alphaq.py]: Should there be `ellipticity_1` for the old version too?

Comment (dkorytov) [reply GCRCatalogs/alphaq.py]: It was wrong any way. Should leave it be wrong the same way?

File: GCRCatalogs/alphaq.py [hunk 4]
```diff
@@ -68,8 +68,8 @@ def _subclass_init(self, filename, **kwargs):
             'stellar_mass':       'totalMassStellar',
             'size_disk_true':     'morphology/diskHalfLightRadiusArcsec',
             'size_bulge_true':    'morphology/spheroidHalfLightRadiusArcsec',
-            'disk_Sersic_index':  'morphology/diskSersicIndex',
-            'bulge_Sersic_index': 'morphology/spheroidSersicIndex',
+            'disk_Sersic':  'morphology/diskSersicIndex',
```
Comment (yymao) [inline GCRCatalogs/alphaq.py]: should be `sersic_disk` (lower case, component name should come after quantity name)

File: GCRCatalogs/alphaq.py [hunk 5]
```diff
@@ -68,8 +68,8 @@ def _subclass_init(self, filename, **kwargs):
...
+            'bulge_Sersic': 'morphology/spheroidSersicIndex',
```
Comment (yymao) [inline GCRCatalogs/alphaq.py]: should be `sersic_bulge` (lower case, component name should come after quantity name)

File: GCRCatalogs/alphaq.py [hunk 6]
```diff
@@ -10,7 +10,7 @@
 from GCR import BaseGenericCatalog
 from distutils.version import StrictVersion
 __all__ = ['AlphaQGalaxyCatalog']
-__version__ = '2.1.1.1' #version 1 for the 2.1.1 catalog reader
+__version__ = '2.1.2.0' #version 1 for the 2.1.1 catalog reader
```
Comment (yymao) [inline GCRCatalogs/alphaq.py]: removing inconsistent inline comment

File: GCRCatalogs/alphaq.py (hunk 4 above)
Comment (dkorytov) [reply GCRCatalogs/alphaq.py]: This google doc has 'disk_Sersic_index' https://docs.google.com/document/d/1rUsImkBkjjw82Xa_-3a8VMV6K9aYJ8mXioaRhz0JoqI/edit#heading=h.9djco1snvvi3

Is there another place with the schema?

**Comment by yymao** [comment]
@dkorytov see https://github.com/LSSTDESC/gcr-catalogs/pull/57

**Comment by dkorytov** [comment]
Got it. :)

**Comment by yymao** [comment]
@dkorytov I've updated your PR (see https://github.com/dkorytov/gcr-catalogs/pull/1). Please take a look and, if everything looks fine, merge it into `dkorytov/gcr-catalogs master` so the changes will reflected here. Thanks.

**Pull request merged**